jupyter_server/extension/application.py

```diff
@@ -169,7 +169,12 @@ def config_file_paths(self):
 
     @classmethod
     def get_extension_package(cls):
-        parts = cls.__module__.split(".")
+        if cls.__module__ == "__main__":
+            # handle direct launch of ExtensionApp module, eg `python -m jupyterlab.labapp`
+            parts = cls.app_name.lower().split(".")
```

Comment:
`app_name` is a property unique to JupyterLab and not something we support here in `jupyter_server`.